wandb/apis/importers/internal.py

```diff
@@ -0,0 +1,437 @@
+        if run_group is not None:
+            run.run_group = run_group
+
+        config = self.run.config()
+        if "_wandb" not in config:
+            config["_wandb"] = {}
+
+        # how do I get this automatically?
+        config["_wandb"]["code_path"] = self.run.code_path()
+        config["_wandb"]["python_version"] = self.run.python_version()
+        config["_wandb"]["cli_version"] = self.run.cli_version()
+
+        self.interface._make_config(
+            data=config,
+            obj=run.config,
+        )  # is there a better way?
+        return self.interface._make_record(run=run)
+
+    def _make_output_record(self, line) -> pb.Record:
+        output_raw = pb.OutputRawRecord()
+        output_raw.output_type = pb.OutputRawRecord.OutputType.STDOUT
+        output_raw.line = line
+        return self.interface._make_record(output_raw=output_raw)
+
+    def _make_summary_record(self) -> pb.Record:
+        d: dict = {
+            **self.run.summary(),
+            "_runtime": self.run.runtime(),  # quirk of runtime -- it has to be here!
+            # '_timestamp': self.run.start_time()/1000,
+        }
+        d = cast_dictlike_to_dict(d)
+        summary = self.interface._make_summary_from_dict(d)
+        return self.interface._make_record(summary=summary)
+
+    def _make_history_records(self) -> Iterable[pb.Record]:
+        for metrics in self.run.metrics():
+            history = pb.HistoryRecord()
+            for k, v in metrics.items():
+                item = history.item.add()
+                item.key = k
+                item.value_json = json.dumps(v)
+            yield self.interface._make_record(history=history)
+
+    def _make_files_record(self, metadata, artifacts, files, media, code) -> pb.Record:
+        files = self.run.files()
+        metadata_fname = f"{self.run_dir}/files/wandb-metadata.json"
+        if files is None:
+            metadata_fname = self._make_metadata_file()
+            files = [(metadata_fname, "end")]
+
+        files_record = pb.FilesRecord()
+        for path, policy in files:
+            if not metadata and path == metadata_fname:
+                continue
+            if not artifacts and path.startswith("artifact/"):
+                continue
+            if not media and path.startswith("media/"):
+                continue
+            if not code and path.startswith("code/"):
+                continue
+
+            f = files_record.files.add()
+            f.path = path
+            f.policy = file_policy_to_enum(policy)  # is this always end?
+
+        return self.interface._make_record(files=files_record)
+
+    def _make_artifact_record(self, artifact, use_artifact=False) -> pb.Record:
+        proto = self.interface._make_artifact(artifact)
+        proto.run_id = self.run.run_id()
+        proto.project = self.run.project()
+        proto.entity = self.run.entity()
+        proto.user_created = use_artifact
+        proto.use_after_commit = use_artifact
```

Comment:
Doesn't this give you an error? It requires setting the current run, see [here](https://github.com/wandb/wandb/blob/395b0dbebaa90f9e9035d4d9d254000da479f6a8/wandb/sdk/internal/internal_api.py#L3230).